sparkmagic/sparkmagic/utils/constants.py

```diff
@@ -76,5 +76,8 @@
                      "Please file an issue at https://github.com/jupyter-incubator/sparkmagic\nError:\n{}"
 EXPECTED_ERROR_MSG = "An error was encountered:\n{}"
 
+YARN_RESOURCE_LIMIT_MSG = "Queue's AM resource limit exceeded."
+RESOURCE_LIMIT_WARNING = "Warning: Session not starting due to resource limitation. {}"
```

Comment:
What about `"Warning: The Spark session does not have enough YARN resources to start up. {}"`

We could then use the `{}` to even provide a link to `/yarnui` in our clusters.